internal/manifests/manifestutils/labels.go

```diff
@@ -26,8 +25,9 @@ import (
 
 func IsFilteredSet(sourceSet string, filterSet []string) bool {
 	for _, pattern := range filterSet {
-		match, _ := regexp.MatchString(pattern, sourceSet)
-		return match
+		if strings.Contains(sourceSet, pattern) {
+			return true
+		}
```

Comment:
Rather than fixing it, it looks more like you're changing the behaviour to not support wildcards at all? That's more a breaking change than a fix.

Reply:
I know it's a break change, but if we think about efficiency and simplicity to end-user, I would say it's better. WDYT?

Reply:
is there a reason this functionality doesn't work as intended today? The user seems to be reporting an issue with just a single filter matching, not multiple so I'm not sure this is going to fix it.

Reply:
I could keep it, but it will keep confusing the users to set the proper regex model to match labels or annotations. Regarding the multiple filters, it was reported [here](https://github.com/open-telemetry/opentelemetry-operator/issues/2770#issuecomment-2031921173).

Reply:
The problem is that the syntax is poorly defined right now. I think the intent was for it to be glob-like, so can we use https://github.com/bmatcuk/doublestar and be done with it? I believe this would keep current semantics and fix the bug.

Reply:
yes, i see the multi-filter issue, but the user is also reporting that single filter isn't working either

Reply:
Right, @jaronoff97, it's because it is confusing how to enable it properly.

Reply:
> https://github.com/bmatcuk/doublestar

I will have a look if we can use it to keep the current semantics.

Reply:
Despite the doublestar being functional, it’s not meeting our requirements for the filters. They check like /*/, but some annotations don’t have any slashes, which the package doesn’t filter properly.
So, I decided to leave the same way, using regexp but using a pre-compilation of the pattern and fixing the multiple annotations.